### ISSUE TITLE
[no-empty-function] should not trigger if empty function is decorated

### ISSUE THREAD

**Issue by JoshMcCullough** [typescript-eslint/typescript-eslint#2838]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

See: https://github.com/JoshMcCullough/eslint-no-empty-function-repro

```ts
  @Log("This is a contrived example.")
  blah(): void { }
```

**Expected Result**

This should _not_ trigger the `@typescript-eslint/no-empty-function` rule because although the function body is empty, it is decorated and a decorator could be doing any number of meaningful/important things.

**Actual Result**

The `@typescript-eslint/no-empty-function` rule is triggered because the decorated function does not have anything within its body.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.9.0` |
| `@typescript-eslint/parser`        | `4.9.0` |
| `TypeScript`                       | `4.1.2` |
| `ESLint`                           | `7.14.0` |
| `node`                             | `11.15.0` |

**Comment by bradzacher** [comment]
happy to accept a PR

**Comment by JoshMcCullough** [comment]
FWIW I'd categorize this as a bug in the rule definition. I haven't contributed here before but will look around and see if it's something that'd be reasonable for me to do. Thanks!

**Comment by a-tarasyuk** [comment]
@bradzacher I think this issue related to the option `decoratedFunctions`. Or am I wrong?
https://github.com/typescript-eslint/typescript-eslint/blob/ccb6b9499a4a4077f2e3d81d0844860a25244a0f/packages/eslint-plugin/tests/rules/no-empty-function.test.ts#L66-L74

**Comment by JoshMcCullough** [comment]
@bradzacher is correct, sorry I did not see the option for `decoratedFunctions`. Adding this config allows empty functions if they're decorated, thanks!

**Comment by a-tarasyuk** [comment]
@JoshMcCullough You are welcome.

**Comment by bradzacher** [comment]
Oops. It's been so long since I touched that rule, and I didn't look at the rule config.
